### ISSUE TITLE
Choosing a YAML schema

### ISSUE THREAD

**Issue by dhimmel** [transpose-publishing/policies-database#3]
We'll want encode the valid YAML structure from #1 into a schema so we can programmatically enforce it. I'm going to use this issue to jot down potential options.

+ https://github.com/23andMe/Yamale
+ http://rx.codesimply.com/
+ https://github.com/Grokzen/pykwalify

**Comment by dhimmel** [comment]
Reusable Data uses the [kwalify](https://github.com/reusabledata/reusabledata/blob/84e84e4a2091bea73f695fd5eff6923d4779e124/.travis.yml#L12-L13) ruby library. It looks like their is a python port of kwalify named [`pykwalify`](https://github.com/Grokzen/pykwalify). `pykwalify` appears to be actively maintained, while Rx and Yamale from above don't appear as active. Hence, I suggest we go with kwalify/pykwalify.

**Comment by jpolka** [comment]
Sounds good. Just so I'm understanding this correctly, the schema would not need to be seen by users - but performs response validation for [records like this one](https://github.com/reusabledata/reusabledata/blob/84e84e4a2091bea73f695fd5eff6923d4779e124/data-sources/biogrid.yaml)?

**Comment by dhimmel** [comment]
> Just so I'm understanding this correctly, the schema would not need to be seen by users - but performs response validation

Exactly. Perhaps users may want to look at it to see what the valid values are for a certain field. So the next step for us will be to convert https://github.com/transpose-publishing/policies-database/issues/1#issuecomment-387247838 into a `schema.yaml` file. @jpolka do you want to try doing that and we can touch base if you get stuck.

**Comment by jpolka** [comment]
I'll give it a shot @dhimmel :) 
What's the benefit of commenting out unset fields? I imagine this might be more confusing to users.

**Comment by dhimmel** [comment]
> It will be best to have you add this file via a pull request. We can go over that tomorrow morning.

@jpolka I will commit this file with your authorship info in a new PR and we can go from there.

**Comment by jpolka** [comment]
Just realized the schema should include a human-readable publisher name. @dhimmel should I create a new branch & pull request on dhimmel/policies-database/schema.yml or...? (thanks in advance)

```yaml
  # Policy/publisher name from SHERAP/RoMEO - "name"
  policy-name:
    type: str
```

**Comment by dhimmel** [comment]
Wait till I'm done with https://github.com/transpose-publishing/policies-database/pull/8 and that's merged.
